src/Compilers/VisualBasic/Portable/Symbols/ConstraintsHelper.vb

```diff
@@ -1086,28 +1091,55 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
             Return New CompoundDiagnosticInfo(diagnostics)
         End Function
 
+        Public Enum ConstructorConstraintError
+            None
+            NoPublicParameterlessConstructor
+            MustInheritType
+            HasRequiredMembers
+        End Enum
+
         ''' <summary>
         ''' Return true if the class type has a public parameterless constructor.
         ''' </summary>
-        Public Function HasPublicParameterlessConstructor(type As NamedTypeSymbol) As Boolean
+        Public Function HasPublicParameterlessConstructor(type As NamedTypeSymbol, synthesizedIfMissing As Boolean) As ConstructorConstraintError
```

Comment:
>, synthesizedIfMissing As Boolean [](http://example.com/codeflow?start=81&length=33)

It looks like we can always infer this value from ```type.TypeKind = TypeKind.Structure```

Reply:
I was mainly shooting for matching the C# implementation, but we can probably simplify the VB impl here.